hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -943,6 +946,33 @@ class IRSuite extends SparkSuite {
     assertEvalsTo(centerColMajor, 13.0)
   }
 
+  @Test def testNDArrayMap() {
+    implicit val execStrats = Set(ExecStrategy.CxxCompile)
+
+    val data = 0 until 10
+    val shape = MakeArray(FastSeq(2L, 5L).map(I64), TArray(TInt64()))
+    val nDim = 2
+
+    def positives = MakeNDArray(nDim, MakeArray(data.map(i => F64(i.toDouble)), TArray(TFloat64())), shape, True())
+    def negatives = NDArrayMap(positives, "e", ApplyUnaryPrimOp(Negate(), Ref("e", TFloat64())))
+    assertEvalsTo(NDArrayRef(positives, MakeArray(FastSeq(1L, 0L), TArray(TInt64()))), 5.0)
+    assertEvalsTo(NDArrayRef(negatives, MakeArray(FastSeq(1L, 0L), TArray(TInt64()))), -5.0)
+
+    def trues = MakeNDArray(nDim, MakeArray(data.map(_ => True()), TArray(TBoolean())), shape, True())
+    def falses = NDArrayMap(trues, "e", ApplyUnaryPrimOp(Bang(), Ref("e", TBoolean())))
+    assertEvalsTo(NDArrayRef(trues, MakeArray(FastSeq(1L, 0L), TArray(TInt64()))), true)
+    assertEvalsTo(NDArrayRef(falses, MakeArray(FastSeq(1L, 0L), TArray(TInt64()))), false)
+
+    def bools = MakeNDArray(nDim,
+      MakeArray(data.map(i => if (i % 2 == 0) True() else False()), TArray(TBoolean())),
```

Comment:
Nitpicky, but I think this test would pass even if the row-major-ness of the map got flipped. Can you check that this doesn't happen?

```
[1 1 1 1 1
 0 0 0 0 0]
vs 
[1 0 1 0 1
 0 1 0 1 0]
```